build_tools/fetch_artifacts.py

```diff
@@ -59,32 +59,38 @@ def retrieve_base_artifacts(args, run_id, build_dir):
         "rocprofiler-sdk_lib",
         "host-suite-sparse_lib",
     ]
-    if args.all or args.blas:
+    if args.blas:
         base_artifacts.append("host-blas_lib")
 
     for base_artifact in base_artifacts:
         s3_exec(GENERIC_VARIANT, base_artifact, run_id, build_dir)
 
 
-def retrieve_enabled_artifacts(args, test_enabled, target, run_id, build_dir):
-    base_artifact_path = []
-    if args.all or args.blas:
-        base_artifact_path.append("blas")
-    if args.all or args.fft:
-        base_artifact_path.append("fft")
-    if args.all or args.miopen:
-        base_artifact_path.append("miopen")
-    if args.all or args.prim:
-        base_artifact_path.append("prim")
-    if args.all or args.rand:
-        base_artifact_path.append("rand")
-    if args.all or args.rccl:
-        base_artifact_path.append("rccl")
+def retrieve_enabled_artifacts(args, target, run_id, build_dir):
+    artifact_paths = []
+    all_artifacts = ["blas", "fft", "miopen", "prim", "rand", "rccl"]
```

Comment:
FYI, if we want this script to work on Windows too, we'll want to exclude some subprojects, like `rccl` here (see https://github.com/ROCm/TheRock/issues/540).